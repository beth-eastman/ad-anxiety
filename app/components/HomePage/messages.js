/*
 * HomePage Messages
 *
 * This contains all the text for the HomePage component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  title: {
    id: 'app.components.HomePage.title',
    defaultMessage: 'Anxiety',
  },
  subtitle: {
    id: 'app.components.HomePage.subtitle',
    defaultMessage: 'Overview',
  },
  header: {
    id: 'app.components.HomePage.page',
    defaultMessage: 'Anxiety comes in many forms. Some people experience anxiety only in specific situations, like when they are asked to speak in public or when they need to fly on an airplane. Others worry a lot, and find themselves feeling anxious on and off throughout the day â€“ and some suffer panic attacks, sudden feelings of anxiety that come on unpredictably. Anxiety is the most common behavioral health problem in the United States, affecting 40 million adults (18.1% of the population aged 18 and older).',
  },
  footer: {
    id: 'app.components.HomePage.bottomPage',
    defaultMessage: 'The Anxiety Assessment, and each of the related assessments, takes about five minutes to complete; recommendations follow based on your results.',
  },
});
