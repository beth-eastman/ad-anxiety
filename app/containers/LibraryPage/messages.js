/*
 * Library Page Messages
 *
 * This contains all the text for the Library Page component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  sectionOne: {
    id: 'app.components.LibraryPage.sectionOne',
    defaultMessage: '1.0 - Introduction to Anxiety',
  },
  sectionTwo: {
    id: 'app.components.LibraryPage.sectionTwo',
    defaultMessage: '2.0 - Anxiety Disorders',
  },
  sectionThree: {
    id: 'app.components.LibraryPage.sectionThree',
    defaultMessage: '3.0 - Other Problems Associated with Anxiety',
  },
  sectionFour: {
    id: 'app.components.LibraryPage.sectionFour',
    defaultMessage: '4.0 - Causes of Anxiety',
  },
  sectionFive: {
    id: 'app.components.LibraryPage.sectionFive',
    defaultMessage: '5.0 - Factors that Make Anxiety Worse',
  },
  sectionSix: {
    id: 'app.components.LibraryPage.sectionSix',
    defaultMessage: '6.0 - Understanding Panic Disorder',
  },
  sectionSeven: {
    id: 'app.components.LibraryPage.sectionSeven',
    defaultMessage: '7.0 - Anxiety Affects Family Members Too',
  },
  sectionEight: {
    id: 'app.components.LibraryPage.sectionEight',
    defaultMessage: '8.0 - How to Manage Anxiety and Worry',
  },
  sectionNine: {
    id: 'app.components.LibraryPage.sectionNine',
    defaultMessage: '9.0 - Getting Support for Anxiety',
  },
  sectionTen: {
    id: 'app.components.LibraryPage.sectionTen',
    defaultMessage: '10.0 - References on Anxiety',
  },
});
