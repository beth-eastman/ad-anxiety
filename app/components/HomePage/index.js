 /*
  * index.js - HomePage
  * The HomePage component contains for the homepage of the application.
  *
  * Created by Bethany Eastman on 06/21/17.
  *
  * AD Anxiety
  *
  * Copyright © 2009-2017 United States Government as represented by
  * the Chief Information Officer of the National Center for Telehealth
  * and Technology. All Rights Reserved.
  *
  * Copyright © 2009-2017 Contributors. All Rights Reserved.
  *
  * THIS OPEN SOURCE AGREEMENT ("AGREEMENT") DEFINES THE RIGHTS OF USE,
  * REPRODUCTION, DISTRIBUTION, MODIFICATION AND REDISTRIBUTION OF CERTAIN
  * COMPUTER SOFTWARE ORIGINALLY RELEASED BY THE UNITED STATES GOVERNMENT
  * AS REPRESENTED BY THE GOVERNMENT AGENCY LISTED BELOW ("GOVERNMENT AGENCY").
  * THE UNITED STATES GOVERNMENT, AS REPRESENTED BY GOVERNMENT AGENCY, IS AN
  * INTENDED THIRD-PARTY BENEFICIARY OF ALL SUBSEQUENT DISTRIBUTIONS OR
  * REDISTRIBUTIONS OF THE SUBJECT SOFTWARE. ANYONE WHO USES, REPRODUCES,
  * DISTRIBUTES, MODIFIES OR REDISTRIBUTES THE SUBJECT SOFTWARE, AS DEFINED
  * HEREIN, OR ANY PART THEREOF, IS, BY THAT ACTION, ACCEPTING IN FULL THE
  * RESPONSIBILITIES AND OBLIGATIONS CONTAINED IN THIS AGREEMENT.
  *
  * Government Agency: The National Center for Telehealth and Technology
  * Government Agency Original Software Designation: ProductName001
  * Government Agency Original Software Title: ProductName
  * User Registration Requested. Please send email
  * with your contact information to: robert.a.kayl.civ@mail.mil
  * Government Agency Point of Contact for Original Software: robert.a.kayl.civ@mail.mil
  *
 */
import React from 'react';
import { Card, CardHeader, CardText } from 'material-ui';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const styles = {
  content: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
};

export default class HomePage extends React.Component {

  /* Change the AppBar title */
  componentWillMount() {
    this.props.setPageTitle("Home"); // eslint-disable-line
    this.props.changeIcon('Home'); // eslint-disable-line
  }

  render() {
    return (
      <div className="homepage">
        <Card style={styles}>
          <CardHeader
            title={<FormattedMessage {...messages.title} />}
            subtitle={<FormattedMessage {...messages.subtitle} />}
          />
          <CardText>
            <FormattedMessage {...messages.header} />
            <br />
            <br />
            <FormattedMessage {...messages.footer} />
          </CardText>
        </Card>
      </div>
    );
  }
}
