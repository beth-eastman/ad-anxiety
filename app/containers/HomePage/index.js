/**
 * Contains content for the homepage of the application.
 */

import React from 'react';
import { Card, CardHeader, CardText } from 'material-ui';
import { FormattedMessage } from 'react-intl';
import messages from './messages';

const styles = {
  content: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
};

export default class HomePage extends React.Component {

  componentWillMount() {
    this.props.setPageTitle("Home"); // eslint-disable-line
  }

  render() {
    return (
      <div className="homepage">
        <Card style={styles}>
          <CardHeader
            title={<FormattedMessage {...messages.title} />}
            subtitle={<FormattedMessage {...messages.subtitle} />}
          />
          <CardText>
            <FormattedMessage {...messages.header} />
            <br />
            <br />
            <FormattedMessage {...messages.footer} />
          </CardText>
        </Card>
      </div>
    );
  }
}
