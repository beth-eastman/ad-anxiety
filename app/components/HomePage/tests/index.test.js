/* eslint-disable */
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import LanguageProvider from 'containers/LanguageProvider';
import { FormattedMessage } from 'react-intl';
import toJson from 'enzyme-to-json';

import HomePage from '../index';
import messages from '../messages';



describe('<HomePage />', () => {
  it('should render the page message', () => {
    const renderedComponent = shallow(
      <MuiThemeProvider>
        <HomePage changeIcon={function () {}} />
      </MuiThemeProvider>
    );

    expect(toJson(renderedComponent)).toMatchSnapshot();
  });
});
