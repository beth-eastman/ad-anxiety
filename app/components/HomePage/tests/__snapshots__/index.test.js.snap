exports[`<HomePage /> should render the page message 1`] = `
<HomePage
  changeIcon={[Function]} />
`;
